### PR TITLE
Fix #4066: shut down executors when `IORuntime.global` shuts down

### LINKED ISSUE #4066: When `IORuntime.global` shuts down, it doesn't shut down the WSTP
... or other `Executor`s. I've noticed this when debugging  #4055. PR incoming...

### PR DESCRIPTION
**Pull request by durban**
Fixes #4066. Related to #4055.

### CODE REVIEW

**Review by armanbilge — APPROVED** [review]
thanks!

File: tests/jvm/src/test/scala/cats/effect/IOAppSpec.scala [hunk 1]
```diff
@@ -335,6 +335,13 @@ class IOAppSpec extends Specification {
           ok
         }
 
+        "shut down WSTP on fatal error without IOApp" in {
+          val h = platform(FatalErrorShutsDownRt, List.empty)
+          h.awaitStatus()
+          h.stdout() must not(contain("sadness"))
```
Comment (djspiewak) [inline tests/jvm/src/test/scala/cats/effect/IOAppSpec.scala:341]: lol

**Review by djspiewak — APPROVED** [review]
Oh I forgot that we completely suppressed the shutdown. So we kinda did this to ourselves. Nice catch!

File: tests/jvm/src/test/scala/cats/effect/IOAppSpec.scala (hunk 1 above)
Comment (durban) [reply tests/jvm/src/test/scala/cats/effect/IOAppSpec.scala:341]: That wasn't me, I've copied it from another test :smile:

**Pull request merged**